### PR TITLE
Rel 5.0.2: Portal Messaging Fix (#3620)

### LINKED ISSUE #3620: 5.0.2 Bug: Patient Portal Secure Mail Reply Function is Broken
### Describe the bug

When you hit the reply button in Secure Mail for the Patient Portal, it doesn't automatically default to the person you are replying to. This happens on both the patient side and the doctor's side. There's also a glitch where only doctors can be replied to with Secure Mail and not other admins who run the clinic (as also shown in below video).

### To Reproduce
[https://www.loom.com/share/f192c49db188434e967c70b736789794](https://www.loom.com/share/f192c49db188434e967c70b736789794)

### Expected behavior
For the reply button to actually work.

### Client configuration
- Browser: Google Chrome Latest
- OpenEMR version: 5.0.2 Latest Patch
- Operating system: 18.04 LTS

### PR DESCRIPTION
**Pull request by tywrenn**
Fixes #3620

#### Short description of what this resolves:
Attempts to fix a portal messaging bug where non-doctors can't be replied to and patients are shown in a way that can crash the browser.

#### Changes proposed in this pull request:
- Change the query on how users are shown
- Change how patients are shown when replying to them

### CODE REVIEW

**Comment by tywrenn** [comment]
@sjpadgett How would I go about adding a find patient box here

**Comment by sjpadgett** [comment]
@tywrenn This is a little hard to do for providers.
- users also include external providers, address book entries(labs etc) so tough to tell what to search for to list.
- I forgot active flag in routine.

Best approach may be to add a flag to users like Portal Auth or add an Acl to test against.

File: portal/messaging/messages.php [hunk 1]
```diff
@@ -58,18 +58,15 @@
...
 }
 
-function getAuthPortalUsers()
-{
+function getAuthPortalUsers() {
     $resultpd = $resultusers = $resultpatients = array();
     if (IS_DASHBOARD) { // admin can mail anyone
-        $authusers = sqlStatement("SELECT users.username as userid,
- CONCAT(users.fname,' ',users.lname) as username, 'user' as type FROM users WHERE authorized = 1");
+        $authusers = sqlStatement("SELECT users.username as userid, CONCAT(users.fname, ' ', users.lname) as username, 'user' as type FROM users WHERE facility_id != 0 AND active = 1");
```
Comment (sjpadgett) [inline portal/messaging/messages.php:65]: Clever but still iffy. I'm going to add a portal provider flag to Users while adding another drop down so, hold off till then(maybe today). Having a separate flag for those allowed to access Portal Dashboard just makes the best sense....

Comment (tywrenn) [reply portal/messaging/messages.php:65]: @sjpadgett Yeah my intention with this change was to allow replies not to just doctors (which would have the auth flag checked) but to also nurses and other people using the EMR. It's weird that we don't have a flag in the system to distinguish them yet if I also do a fix for #3561 that wouldn't need to be created.

**Comment by tywrenn** [comment]
I'll have to create another pull for this that does this for v6

**Comment by tywrenn** [comment]
@bradymiller Just noticed the milestone for 5.0.3 hasn't changed to 6.0.0 -- might wanna change this

**Comment by bradymiller** [comment]
@tywrenn , thanks for pointing out the milestone issue. Just fixed that.

**Pull request closed without merging**